File: AndroidDemoProject/app/src/main/java/com/by/androiddemoproject/bridges/flutter/FlutterMethodChannelBridge.kt
package com.by.androiddemoproject.bridges.flutter

// MARK: - MethodChannel Bridge documentation
// In a real project, register these in FlutterContainerActivity.configureFlutterEngine():
//
// val eventChannel = MethodChannel(flutterEngine.dartExecutor.binaryMessenger, "com.app.eventbus")
// eventChannel.setMethodCallHandler { call, result ->
//     when (call.method) {
//         "sendMessage" -> {
//             val json = call.arguments as? String ?: return@setMethodCallHandler
//             val message = EventMessage.fromJSONString(json) ?: run {
//                 result.error("INVALID", "Invalid message JSON", null)
//                 return@setMethodCallHandler
//             }
//             EventBus.dispatch(message)
//             result.success(null)
//         }
//     }
// }
//
// val routerChannel = MethodChannel(flutterEngine.dartExecutor.binaryMessenger, "com.app.router")
// routerChannel.setMethodCallHandler { call, result ->
//     when (call.method) {
//         "navigate" -> {
//             val url = call.arguments as? String ?: return@setMethodCallHandler
//             AppRouter.navigate(context, url)
//             result.success(null)
//         }
//         "pop" -> {
//             finish()
//             result.success(null)
//         }
//     }
// }
//
// To send TO Flutter:
// eventChannel.invokeMethod("onEventMessage", message.toJSONString())

object FlutterMethodChannelBridge {
    const val EVENT_CHANNEL_NAME = "com.app.eventbus"
    const val ROUTER_CHANNEL_NAME = "com.app.router"
}

File: iOSDemoProject/iOSDemoProject/Bridges/FlutterBridge/FlutterMethodChannelBridge.swift
import Foundation

// MARK: - Flutter MethodChannel Bridge (placeholder for actual Flutter integration)
///
/// In a real project, this registers MethodChannel handlers on the FlutterEngine:
///
/// ```swift
/// let eventChannel = FlutterMethodChannel(name: "com.app.eventbus", binaryMessenger: engine.binaryMessenger)
/// eventChannel.setMethodCallHandler { call, result in
///     if call.method == "sendMessage", let json = call.arguments as? String {
///         guard let message = EventMessage.from(jsonString: json) else {
///             result(FlutterError(code: "INVALID", message: "Invalid message JSON", details: nil))
///             return
///         }
///         EventBus.shared.dispatch(message: message)
///         result(nil)
///     }
/// }
///
/// let routerChannel = FlutterMethodChannel(name: "com.app.router", binaryMessenger: engine.binaryMessenger)
/// routerChannel.setMethodCallHandler { call, result in
///     if call.method == "navigate", let url = call.arguments as? String {
///         AppRouter.shared.navigate(to: url)
///         result(nil)
///     } else if call.method == "pop" {
///         AppRouter.shared.pop()
///         result(nil)
///     }
/// }
/// ```
///
/// To send messages TO Flutter:
/// ```swift
/// eventChannel.invokeMethod("onEventMessage", arguments: message.toJSONString())
/// ```
class FlutterMethodChannelBridge {
    static let eventChannelName = "com.app.eventbus"
    static let routerChannelName = "com.app.router"
}
